### ISSUE TITLE
MA0045 shouldn't trigger on ADO.NET SQLite

### ISSUE THREAD

**Issue by Sella-GH** [meziantou/Meziantou.Analyzer#1121]
### Version of the Meziantou.Analyzer NuGet package

3.0.63

### Rule Identifier

MA0045

### Target Framework

net10.0

### C# Language version

C# 14

### Description

Per definition of the [Async limitations](https://learn.microsoft.com/en-us/dotnet/standard/data/sqlite/async) document in the docs, `async` methods should be avoided due to lack of asynchronous I/O support in SQLite.

### Reproduction Steps

Minimal code:

```c#
private static void Execute(SqliteConnectionStringBuilder connectionString)
{
    using SqliteConnection connection = new(connectionString.ConnectionString);
    connection.Open(); // Use 'OpenAsync' instead

    using SqliteCommand command = connection.CreateCommand(); // Prefer using 'await using' (probably reasonable)
    command.CommandText = "Some SQL Command";
    command.ExecuteNonQuery(); // Use 'ExecuteNonQueryAsync' instead
}
```

### Other information
Maybe related to #1117

**Comment by meziantou** [comment]
The main issue is that there is no guarantee on the actual type because the Sqlite classes are not sealed.

```
SqliteCommand command = new CustomSqliteCommand();
command.ExecuteNonQuery(); // This could actually benefit from async if the custom class implement it correctly.
```

SqlConnection.CreateCommand can be special-case, but this is not safe to special case other methods without more complex code. You would need to analyze how the value is built to be sure of the type. Data flow analysis is more complex and more CPU intensive.

**Comment by meziantou** [comment]
I think a good tradeoff would be to add an option to opt-in or opt-out SqliteConnection/SqliteCommand methods

**Comment by Sella-GH** [comment]
If it's too compute intensive then the tradeoff with the opt-in switch would be good. Could at least silence the analyzer for the time being.

**Comment by meziantou** [comment]
The new option is opt-out, so after updating to the latest version (available very soon on nuget.org), sqlite async methods should now be excluded.

**Comment by Sella-GH** [comment]
Thanks! The detection is mostly gone, however `SqliteConnection.Open()` is still flagged. Was that an oversight or an intentional decision?

**Comment by meziantou** [comment]
Fix by https://github.com/meziantou/Meziantou.Analyzer/pull/1129

**Comment by Sella-GH** [comment]
Thank you! This is fixed now 🙂

**Comment by Sella-GH** [comment]
Ugh, I assume we're not done yet. Just found some more:
- `SqliteCommand.ExecuteReader`
- `SqliteDataReader.Read`
- `SqliteDataReader.IsDBNull`

I suspect the easier way would be to exclude the whole assembly? Would save much additional code and branches that way probably..

**Comment by meziantou** [comment]
https://github.com/meziantou/Meziantou.Analyzer/pull/1130 Extends all methods from SqliteConnection, SqliteCommand, and SqliteDataReader
